### PR TITLE
Avoid manual initialisations of system or user configuration

### PR DESCRIPTION
**Pull request by Alkarex**
More consistent use of Context. Code reduction.
Moved the complicated configuration initialisations to the Context object.
Promised in https://github.com/FreshRSS/FreshRSS/pull/3003#discussion_r439742363

### CODE REVIEW

File: lib/Minz/FrontController.php [hunk 1]
```diff
@@ -31,9 +31,6 @@ class Minz_FrontController {
 	 */
 	public function __construct () {
 		try {
-			Minz_Configuration::register('system',
-			                             DATA_PATH . '/config.php',
-			                             FRESHRSS_PATH . '/config.default.php');
```
Comment (Alkarex) [inline lib/Minz/FrontController.php:36]: @marienfressinaud It was not so nice to initialise the system configuration here with a FreshRSS constant in Minz. What do you think?

File: app/Models/Context.php [hunk 2]
```diff
@@ -76,6 +98,14 @@ public static function init() {
 		}
 	}
 
+	/**
+	 * Initialize the context for the global system and the current user
+	 */
+	public static function init() {
```
Comment (Alkarex) [inline app/Models/Context.php]: Deprecated. We can maybe remove it immediately.

File: lib/Minz/FrontController.php (hunk 1 above)
Comment (marienfressinaud) [reply lib/Minz/FrontController.php:36]: I agree, an alternative would be to rename the constant to something like `ROOT_PATH` (or create one)

**Comment by Alkarex** [comment]
@aledeg A bit of help to finalise this PR would be much welcome, if you are motivated enough :-)

In any case, the idea is simply to use `FreshRSS_Context` everywhere instead of a bit randomly as we did before.
The status is that it is almost done but needs some testing, corresponding fixes, and proof reading. As it spans over many files, the risk of conflicts with new PRs is high, so it would be nice to merge it.

**Comment by Alkarex** [comment]
I am looking a bit more at it now

**Comment by aledeg** [comment]
I've looked quickly at it (did not test) and it looks good to me.
I have a question though. Do you need to keep `FreshRSS_ConfigurationSetter` as there is only one call to it and you removed it?
I also notice that you sometimes use `==` and sometimes `===`. Is there a reason for that?

Apart from that, what help do you need?

**Comment by Alkarex** [comment]
Ok, I will look at `FreshRSS_ConfigurationSetter`. The `== ''` is when I want to catch more largely (empty string, null, false...).
I think it is almost ready for a test (I have just found another little bug in the install process - I will change the status of the PR when fixed).
What would be nice would mainly be proof reading and tests (positive = everything should work, but also negative = denying access when not legitimate) of the different entry points and login types / variants (anonymous access...), and things related to user management (i.e. admin aspects such as creating users).

**Comment by aledeg** [comment]
OK, let me know when ready.

**Comment by Alkarex** [comment]
Ready, but best to merge https://github.com/FreshRSS/FreshRSS/pull/3305 first

File: app/Controllers/errorController.php [hunk 3]
```diff
@@ -33,10 +33,12 @@ public function indexAction() {
 		case 500:
 			header('HTTP/1.1 500 Internal Server Error');
 			$this->view->code = 'Error 500 - Internal Server Error';
+			$this->view->errorMessage = 'Error 500 - Internal Server Error';
```
Comment (Alkarex) [inline app/Controllers/errorController.php:36]: The view is using `errorMessage` (should have been another PR)

**Comment by Alkarex** [comment]
I running this on my main instance; seems stable

**Comment by Alkarex** [comment]
Looks all good in my tests. Let's continue testing in master 🤞🏻

**Pull request merged**